src/nnet3bin/nnet3-egs-augment-image.cc

```diff
@@ -275,6 +292,68 @@ void PerturbImage(const ImageAugmentationConfig &config,
   ApplyAffineTransform(transform_mat, config.num_channels, image, fill_mode);
 }
 
+/**
+  This function scales the shortest size to a size determined by 'config'
+  It also crops the image to a determined size
+*/
+void ScaleAndCropImage(const ImageAugmentationConfig &config,
+                Matrix<BaseFloat> *image) {
+  int32 width = image->NumRows(),
+      num_channels = config.num_channels,
+      height = image->NumCols() / num_channels,
+      crop_size = config.crop_size;
+  FillMode fill_mode = config.GetFillMode();
+  if (image->NumCols() % num_channels != 0) {
+    KALDI_ERR << "Number of Cols must be divisible by number of channels";
+  }
+  
+  Matrix<BaseFloat> scale_mat(3, 3, kUndefined);
+  scale_mat.SetUnit();
+  int32 scale = RandInt(config.crop_scale_min,config.crop_scale_max);
+  //int32 scale = 256;
+  if (width > height) {
+    scale_mat(0,0) = (scale * 1.0) / height;
+    scale_mat(1,1) = (scale * 1.0) / height;
+  } else {
+    scale_mat(0,0) = (scale * 1.0) / width;
+    scale_mat(1,1) = (scale * 1.0) / width;
+  }
+
+  int32 new_width = static_cast<int32>(width * scale_mat(0,0));
+  int32 new_height = static_cast<int32>(height * scale_mat(1,1));
+  int32 start_row = RandInt(0,new_width-crop_size);
```

Comment:
Please use a space after the comma, here and everywhere else. Also, use spaces between arithmetic operators/operands (i.e. `new_width - crop_size`).